ibis/backends/bigquery/__init__.py

```diff
@@ -528,8 +529,48 @@ def list_databases(self, like=None):
         return self.list_schemas(like=like)
 
     def list_tables(
-        self, like: str | None = None, database: str | None = None
+        self,
+        like: str | None = None,
+        database: str | None = None,
+        schema: str | None = None,
     ) -> list[str]:
+        """List the tables in the database.
+
+        Parameters
+        ----------
+        like
+            A pattern to use for listing tables.
+        database
+            The database (project) to perform the list against.
+        schema
+            The schema (dataset) inside `database` to perform the list against.
+
+            ::: {.callout-warning}
+            ## `schema` refers to database hierarchy
+
+            The `schema` parameter does **not** refer to the column names and
+            types of `table`.
+            :::
+        """
+        if database is not None and schema is None:
+            util.warn_deprecated(
+                "database",
+                instead=(
+                    f"{self.name} cannot list tables only using `database` specifier. "
+                    "Include a `schema` argument."
+                ),
+                as_of="7.1",
+                removed_in="8.0",
+            )
+            database = sg.parse_one(database, into=sg.exp.Table).sql(dialect=self.name)
+        elif database is None and schema is not None:
+            database = sg.parse_one(schema, into=sg.exp.Table).sql(dialect=self.name)
+        else:
+            database = (
+                sg.table(schema, db=database, quoted=True).sql(dialect=self.name)
```

Comment:
`quoted=True` shouldn't be necessary here. `sg.table` will automatically determine whether the identifier needs to be quoted.